### ISSUE TITLE
python-casacore errors when installing on MacOS 10.13.6

### ISSUE THREAD

**Issue by vdhulst** [casacore/python-casacore#163]
When building python-casacore I get the following errors (all connected to two casacore include files). I did install the latest version of casacore successfully first. The error messages are:
```
In file included from src/fit.cc:31:
In file included from /usr/local/include/casacore/python/Converters/PycBasicData.h:34:
In file included from /usr/local/include/casacore/python/Converters/PycArray.h:36:
In file included from /usr/local/include/casacore/casa/Arrays/Array.h:35:
In file included from /usr/local/include/casacore/casa/Arrays/ArrayBase.h:36:
/usr/local/include/casacore/casa/Containers/Allocator.h:62:12: error: unknown
      type name 'constexpr'
  explicit constexpr ArrayInitPolicy(bool v): init(v) {}
           ^
/usr/local/include/casacore/casa/Containers/Allocator.h:62:22: error: 
      constructor cannot have a return type
  explicit constexpr ArrayInitPolicy(bool v): init(v) {}
                     ^~~~~~~~~~~~~~~
/usr/local/include/casacore/casa/Containers/Allocator.h:68:12: error: unknown
      type name 'constexpr'
    static constexpr ArrayInitPolicy NO_INIT = ArrayInitPolicy(false);
           ^
/usr/local/include/casacore/casa/Containers/Allocator.h:68:37: error: expected
      ';' at end of declaration list
    static constexpr ArrayInitPolicy NO_INIT = ArrayInitPolicy(false);
                                    ^
/usr/local/include/casacore/casa/Containers/Allocator.h:70:12: error: unknown
      type name 'constexpr'
    static constexpr ArrayInitPolicy INIT = ArrayInitPolicy(true);
           ^
/usr/local/include/casacore/casa/Containers/Allocator.h:70:37: error: expected
      ';' at end of declaration list
    static constexpr ArrayInitPolicy INIT = ArrayInitPolicy(true);
                                    ^
/usr/local/include/casacore/casa/Containers/Allocator.h:88:10: error: unknown
      type name 'constexpr'
  static constexpr size_t alignment = ALIGNMENT;
         ^
/usr/local/include/casacore/casa/Containers/Allocator.h:88:26: error: expected
      ';' at end of declaration list
  static constexpr size_t alignment = ALIGNMENT;
                         ^
/usr/local/include/casacore/casa/Containers/Allocator.h:97:53: error: expected
      ';' at end of declaration list
  casacore_allocator(const casacore_allocator&other) noexcept
                                                    ^
/usr/local/include/casacore/casa/Containers/Allocator.h:152:22: error: expected
      ';' at end of declaration list
  new_del_allocator() noexcept {
                     ^
In file included from src/fit.cc:31:
In file included from /usr/local/include/casacore/python/Converters/PycBasicData.h:34:
In file included from /usr/local/include/casacore/python/Converters/PycArray.h:36:
In file included from /usr/local/include/casacore/casa/Arrays/Array.h:36:
In file included from /usr/local/include/casacore/casa/Containers/Block.h:34:
In file included from /usr/local/include/casacore/casa/Utilities/Assert.h:32:
/usr/local/include/casacore/casa/Exceptions/Error.h:146:23: error: exception
      specification of overriding function is more lax than base version
  virtual const char* what() const noexcept
                      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/exception:102:25: note: 
      overridden virtual function is here
    virtual const char* what() const _NOEXCEPT;
                        ^
In file included from src/fit.cc:31:
In file included from /usr/local/include/casacore/python/Converters/PycBasicData.h:34:
In file included from /usr/local/include/casacore/python/Converters/PycArray.h:36:
In file included from /usr/local/include/casacore/casa/Arrays/Array.h:36:
In file included from /usr/local/include/casacore/casa/Containers/Block.h:34:
In file included from /usr/local/include/casacore/casa/Utilities/Assert.h:32:
/usr/local/include/casacore/casa/Exceptions/Error.h:146:35: error: expected ';'
      at end of declaration list
  virtual const char* what() const noexcept
                                  ^
/usr/local/include/casacore/casa/Exceptions/Error.h:172:3: error: exception
      specification of overriding function is more lax than base version
  ~AipsError() noexcept;
  ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/exception:101:13: note: 
      overridden virtual function is here
    virtual ~exception() _NOEXCEPT;
            ^
In file included from src/fit.cc:31:
In file included from /usr/local/include/casacore/python/Converters/PycBasicData.h:34:
In file included from /usr/local/include/casacore/python/Converters/PycArray.h:36:
In file included from /usr/local/include/casacore/casa/Arrays/Array.h:36:
In file included from /usr/local/include/casacore/casa/Containers/Block.h:34:
In file included from /usr/local/include/casacore/casa/Utilities/Assert.h:32:
/usr/local/include/casacore/casa/Exceptions/Error.h:172:15: error: expected ';'
      at end of declaration list
  ~AipsError() noexcept;
              ^
/usr/local/include/casacore/casa/Exceptions/Error.h:253:16: error: expected ';'
      at end of declaration list
  ~AllocError() noexcept;
               ^
/usr/local/include/casacore/casa/Exceptions/Error.h:294:16: error: expected ';'
      at end of declaration list
  ~IndexError() noexcept;
               ^
/usr/local/include/casacore/casa/Exceptions/Error.h:336:16: error: expected ';'
      at end of declaration list
  ~indexError() noexcept;
               ^
/usr/local/include/casacore/casa/Exceptions/Error.h:376:15: error: expected ';'
      at end of declaration list
  ~DuplError() noexcept;
              ^
/usr/local/include/casacore/casa/Exceptions/Error.h:419:15: error: expected ';'
      at end of declaration list
  ~duplError() noexcept;
              ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
```

Hope there is a simple solution. I will be in Dwingeloo next week Tuesday, so could discuss this in person then with Ger or Gijs.

**Comment by gijzelaerr** [comment]
What compiler are you using?

**Comment by ludwigschwardt** [comment]
I just ran into this while reviving the casacore homebrew formula :-)

My solution was to specify C++11 when installing the Python package, e.g.
```
CPPFLAGS='-std=c++11' pip install python-casacore
```
@gijzelaerr, in Homebrew we can optionally disable C++11 but I guess that is suboptimal... Can we add this flag to the extension builder in python-casacore's `setup.py`?

**Comment by gijzelaerr** [comment]
sure!

**Comment by gijzelaerr** [comment]
@ludwigschwardt i lost track, has this issue been resolved?

**Comment by ludwigschwardt** [comment]
Fixed in #173.
